src/test/scala/org/graphframes/lib/ConnectedComponentsSuite.scala

```diff
@@ -105,20 +181,21 @@ class ConnectedComponentsSuite extends SparkFunSuite with GraphFrameTestSparkCon
     val components1 = cc.setCheckpointInterval(1).run()
     assertComponents(components1, expected)
     assert(isFromCheckpoint(components1),
-      "The result should depend on chekpoint data if checkpoint interval is 1.")
+      "The result should depend on checkpoint data if checkpoint interval is 1.")
 
     val components10 = cc.setCheckpointInterval(10).run()
     assertComponents(components10, expected)
     assert(!isFromCheckpoint(components10),
       "The result shouldn't depend on checkpoint data if converged before first checkpoint.")
   }
 
-  private def assertComponents[T: ClassTag](actual: DataFrame, expected: Set[Set[T]]): Unit = {
-    // note: not using agg + collect_list because collect_list is not available in 1.6.2
-    val actualComponents = actual.select("component", "id").rdd
-      .map { case Row(component: Long, id: T) =>
-        (component, id)
-      }.groupByKey()
+  private def assertComponents[T: ClassTag:TypeTag](
+      actual: DataFrame,
+      expected: Set[Set[T]]): Unit = {
+    import actual.sqlContext.implicits._
+    // note: not using agg + collect_list because collect_list is not available in 1.6.2 w/o hive
+    val actualComponents = actual.select("component", "id").as[(Long, T)].rdd
```

Comment:
Switched to `as` because in Spark 1.6 it returns `java.lang.Long` instead of `long`, which messed up case matching.